api/v1alpha1/ionoscloudmachine_types.go

```diff
@@ -70,9 +85,17 @@ type IONOSVolumeSpec struct {
 	SSHKeys *[]string `json:"sshKeys,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule="(has(self.useDHCP) && !has(self.primaryAddressesFrom)) || (!has(self.useDHCP) && has(self.primaryAddressesFrom))", message="You need to specify either useDHCP or primaryAddressesFrom"
 type IONOSNicSpec struct {
 	LanRef    IONOSLanRefSpec `json:"lanRef"`
 	PrimaryIP *string         `json:"primaryIP,omitempty"`
+	// PrimaryAddressesFrom is an IPAddressPools that should be assigned
+	// to an IPAddressClaims.
+	// +optional
+	PrimaryAddressesFrom *corev1.TypedLocalObjectReference `json:"primaryAddressesFrom,omitempty"`
+	// +kubebuilder:default:=true
+	// +kubebuilder:validation:Optional
+	UseDHCP bool `json:"useDHCP"`
```

Comment:
Your are sure this true? Otherwise use nullable.

Reply:
not used anymore